tests/unit/test_hnsw.cpp

```diff
@@ -2093,13 +2097,14 @@ TYPED_TEST(HNSWTest, markDelete) {
         }
     }
 
-    // Unmark the previously marked vectors.
-    for (labelType label = 0; label < n; label++)
-        if (label % 2 == ep_reminder)
-            this->CastToHNSW(index)->unmarkDelete(label);
+    // Re-add the previously marked vectors (under new internal ids).
+    for (labelType label = 0; label < n; label++) {
+        if (label % 2 == ep_reminder) {
+            GenerateAndAddVector<TEST_DATA_T>(index, dim, label, label);
+        }
+    }
 
-    ASSERT_EQ(this->CastToHNSW(index)->getNumMarkedDeleted(), 0);
-    ASSERT_EQ(VecSimIndex_IndexSize(index), n + 1);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), n + n / 2 + 1);
```

Comment:
also add validation that the number of the vectors that are marked as deleted is still n/2